tests/components/shelly/test_binary_sensor.py

```diff
@@ -39,12 +39,16 @@ async def test_block_binary_sensor(
     entity_id = f"{BINARY_SENSOR_DOMAIN}.test_name_channel_1_overpowering"
     await init_integration(hass, 1)
 
-    assert hass.states.get(entity_id).state == STATE_OFF
+    state = hass.states.get(entity_id)
```

Comment:
This can be a single line, we encourage using the walrus operator in core and there I found the same in 62 tests mostly newer ones, so I assume it is encouraged in tests also:
```suggestion
    assert (state := hass.states.get(entity_id)) and state.state == STATE_OFF
```
in other places if the line is too long we can split int two lines, but I don't see the benefit of 3 lines as it is now.
```python
    assert (state := hass.states.get(entity_id))
    assert state.state == STATE_OFF
```

Reply:
The idea was to align the style/syntax on all lines.
Also, it seems to me that the walrus is not used on any other integration tests.

Reply:
As I wrote at https://github.com/home-assistant/core/pull/140080#discussion_r1994432052 there are 62 files with walrus 
operator in tests, 
You can run the search yourself to check: [search for "assert (state := hass.states.get"](https://github.com/search?q=repo%3Ahome-assistant%2Fcore+%22assert+%28state+%3A%3D+hass.states.get%22&type=code
)

I think this example:
https://github.com/home-assistant/core/blob/f801cfee7e59d917cbb73fc30af05695b249502e/tests/components/wled/test_switch.py#L114-L115
which is for an integration that @frenck  is a code owner is a good example, why we want to encourage this.